src/lib/socket.c

```diff
@@ -57,6 +57,42 @@
 #include <openssl/ssl.h>
 #include <openssl/err.h>
 #include <openssl/bio.h>
+#if OPENSSL_VERSION_NUMBER < 0x10100001L
```

Comment:
libressl has frozen OPENSSL_VERSION_NUMBER. The feature conditionals should be split out to HAVE_XXX so they can be defined separately.

This stuff just screams for autoconf....